src/com/puppetlabs/puppetdb/config.clj

```diff
@@ -208,13 +211,20 @@
     (s/validate command-processing-out converted-config)
     (assoc config :command-processing converted-config)))
 
+(defn configure-puppetdb
+  "Validates the [puppetdb] section of the config"
+  [{:keys [puppetdb] :as config :or {puppetdb {}}}]
```

Comment:
Would it make sense to use s/maybe in puppetdb-config-in rather than defaulting to an empty map here? Defaulting like this seems weird to me but I'm not sure of the downstream implications.